### PR TITLE
feat(hp): Add displayArea option for Hanging protocols and example with Mamo

### PR DESCRIPTION
**Pull request by wayfarer3130**
### Context

Add a mammography hanging protocol, mostly as a test/demonstration code.

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] OS: 
- [] Node version: 
- [] Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: extensions/default/src/hangingprotocols/mammoDisplaySetSelector.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
+const priorStudyMatchingRules = [
+  {
+    // The priorInstance is a study counter that indicates what position this study is in
+    // and the value comes from the options parameter.
+    attribute: 'studyInstanceUIDsIndex',
+    from: 'options',
```
Comment (sedghi) [inline extensions/default/src/hangingprotocols/mammoDisplaySetSelector.ts:6]: I was looking at the options documentation, it is not clear, it says this 

```md
### `from` attribute
The from attribute allows getting the attribute to test from some other object
such as the prior study, the list of studies overall or another module provided
value.  Some of the possible attributes are:

* `prior`: To get the value from the prior study.
* `activeStudy`: To match the active study
* `studies`: To match the list of studies to display
* `displaySets`: The display sets for the current study
* `allDisplaySets`: Alll available display sets
* `instance`: An instance from the current display set being tested
* `options`: Gets the options object itself, eg if you want a simple top level
  value.
```

But it is still not clear at first sight where the options are provided and by whom.

Comment (sedghi) [reply extensions/default/src/hangingprotocols/mammoDisplaySetSelector.ts:6]: Can you please update that documentation with extra details?

File: extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts [hunk 2]
```diff
@@ -360,6 +362,15 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
       if (camera) {
         viewport.setCamera(camera);
       }
+      const pan = viewport.getPan();
+      if (flip) {
+        viewport.setCamera({ flipHorizontal: true });
+      }
```
Comment (sedghi) [inline extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts:368]: Why do you assume `flip` means flipHorizontal, maybe rename it to flipHorizontal then in the HP?

File: extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts [hunk 3]
```diff
@@ -585,12 +587,21 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
...
-      }
+    }
     }
 
     return viewport.setStack(imageIds, initialImageIndexToUse).then(() => {
       viewport.setProperties({ ...properties });
       this.setPresentations(viewport.id, presentations);
+      if (displayArea) {
+        viewport.setOptions({ displayArea }, true);
+      }
```
Comment (sedghi) [inline extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts:598]: shouldn't we use the setDisplayArea here?

**Review by sedghi — APPROVED** [review]

**Comment by ranasrule** [comment]
Can you post a link to the Mamo example?

**Pull request merged**